### ISSUE TITLE
Install is confused (from #613)

### ISSUE THREAD

**Issue by mih** [datalad/datalad#834]
````
/tmp % datalad install ///
2016-09-15 13:14:39,283 [INFO   ] Installing /tmp/datasets.datalad.org from http://datasets.datalad.org/ (install.py:389)
2016-09-15 13:14:39,283 [INFO   ] Install dataset at: /tmp/datasets.datalad.org (install.py:426)
1 installed item is available at
<Dataset path=/tmp/datasets.datalad.org>
/tmp/datasets.datalad.org (git)-[master] % datalad install -d . -r labs
2016-09-15 13:20:43,199 [WARNING] received 'None' as 'source'. (install.py:55)
2016-09-15 13:20:43,199 [INFO   ] Installing /tmp/datasets.datalad.org/labs (install.py:389)
2016-09-15 13:20:43,199 [INFO   ] Install subdataset at: /tmp/datasets.datalad.org/labs (install.py:407)
2016-09-15 13:20:43,476 [ERROR  ] stderr| Cloning into 'labs'...
| fatal: repository 'http://datasets.datalad.org/.git/labs/' not found
| fatal: clone of 'http://datasets.datalad.org/.git/labs' into submodule path 'labs' failed (cmd.py:352)
2016-09-15 13:20:43,477 [ERROR  ] Failed to run ['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'submodule', 'update', '--checkout', '--init', '--', 'labs'] under '/tmp/datasets.datalad.org'. Exit code=128. out=Submodule 'labs' (http://datasets.datalad.org/.git/labs) registered for path 'labs'
|  err=Cloning into 'labs'...
| fatal: repository 'http://datasets.datalad.org/.git/labs/' not found
| fatal: clone of 'http://datasets.datalad.org/.git/labs' into submodule path 'labs' failed
|  (cmd.py:296)
2016-09-15 13:20:43,480 [ERROR  ] CommandError: command '['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'submodule', 'update', '--checkout', '--init', '--', 'labs']' failed with exitcode 128.
| Failed to run ['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'submodule', 'update', '--checkout', '--init', '--', 'labs'] under '/tmp/datasets.datalad.org'. Exit code=128. out=Submodule 'labs' (http://datasets.datalad.org/.git/labs) registered for path 'labs'
|  err=Cloning into 'labs'...
| fatal: repository 'http://datasets.datalad.org/.git/labs/' not found
| fatal: clone of 'http://datasets.datalad.org/.git/labs' into submodule path 'labs' failed
|  [cmd.py:run:297] (CommandError) (main.py:259)
````

**Comment by mih** [comment]
and 

````
/tmp/datasets.datalad.org (git)-[master] % git st
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	deleted:    labs

no changes added to commit (use "git add" and/or "git commit -a")
````

**Comment by bpoldrack** [comment]
Fixed.

**Comment by yarikoptic** [comment]
confirming -- is fixed with #613
